# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [1]:
salary = float(input("Enter your salary: "))
years = int(input("Enter years of service: "))

if years > 5:
    bonus = salary * 5 / 100
    print("Your bonus is:", bonus)
else:
    print("You are not eligible for a bonus.")

Your bonus is: 2500.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [2]:
age = int(input("Enter your age: "))

if age > 17:
    print("You are eligible for voting.")
else:
    print("You are not eligible for voting.")

You are eligible for voting.



3) Write a program to check whether a number entered by user is even or odd.

In [3]:
number = int(input("Enter a number: "))

if number % 2 == 0:
    print(number, "is an Even number.")
else:
    print(number, "is an Odd number.")


8 is an Even number.


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [4]:
number = int(input("Enter a number: "))

if number % 7 == 0:
    print(number, "is divisible by 7.")
else:
    print(number, "is not divisible by 7.")
    

49 is divisible by 7.


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [6]:
number = int(input("Enter a number: "))

if number % 5 == 0:
    print("Hello")
else:
    print("Bye")

Bye


7) Write a program to display the last digit of a number.


In [8]:
number = input("Enter a number: ")

print("Last digit is:", number[-1])


Last digit is: 9


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [9]:
length = float(input("Enter length: "))
breadth = float(input("Enter breadth: "))

if length == breadth:
    print("It is a Square.")
else:
    print("It is a Rectangle.")

It is a Square.


10) Take two int values from user and print greatest among them.

In [10]:
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

if num1 > num2:
    print(num1, "is the greatest.")
elif num1 == num2:
    print("Both numbers are equal.")
else:
    print(num2, "is the greatest.")

85 is the greatest.


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [12]:
quantity = int(input("Enter quantity: "))

cost = quantity * 100

if cost > 1000:
    discount = cost * 10 / 100
    total = cost - discount
    print("Original cost:", cost)
    print("Discount:", discount)
    print("Total cost after discount:", total)
else:
    print("No discount applied.")
    print("Total cost:", cost)

Original cost: 1500
Discount: 150.0
Total cost after discount: 1350.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [13]:
marks = int(input("Enter your marks: "))

if marks < 25:
    print("Grade: F")
elif marks >= 25 and marks <= 45:
    print("Grade: E")
elif marks > 45 and marks <= 50:
    print("Grade: D")
elif marks > 50 and marks <= 60:
    print("Grade: C")
elif marks > 60 and marks <= 80:
    print("Grade: B")
else:
    print("Grade: A")

Grade: A


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [14]:
total_classes = int(input("Enter number of classes held: "))
attended_classes = int(input("Enter number of classes attended: "))

percentage = (attended_classes / total_classes) * 100

print("Attendance percentage:", percentage, "%")

if percentage >= 75:
    print("Student is allowed to sit in exam.")
else:
    print("Student is NOT allowed to sit in exam.")

Attendance percentage: 60.0 %
Student is NOT allowed to sit in exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [18]:
total_classes = int(input("Enter number of classes held: "))
attended_classes = int(input("Enter number of classes attended: "))

percentage = (attended_classes / total_classes) * 100

print("Attendance percentage:", percentage, "%")

if percentage >= 75:
    print("Student is allowed to sit in exam.")
else:
    medical = input("Do you have a medical cause? (Y/N): ")
    if medical == "Y" or medical == "y":
        print("Student is allowed to sit in exam due to medical cause.")
    else:
        print("Student is NOT allowed to sit in exam.")

Attendance percentage: 25.0 %
Student is allowed to sit in exam due to medical cause.


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [19]:
year = int(input("Enter a year: "))

if year % 4 == 0 and year % 100 != 0:
    print(year, "is a Leap Year.")
elif year % 400 == 0:
    print(year, "is a Leap Year.")
else:
    print(year, "is NOT a Leap Year.")

2000 is a Leap Year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [22]:
age = int(input("Enter your age: "))
gender = input("Enter gender (M/F): ")
marital_status = input("Enter marital status (Y/N): ")

if gender == "F" or gender == "f":
    print("Place of service: Urban Areas")
elif gender == "M" or gender == "m":
    if age >= 20 and age <= 40:
        print("Place of service: Anywhere")
    elif age > 40 and age <= 60:
        print("Place of service: Urban Areas")
    else:
        print("ERROR")
else:
    print("ERROR")

Place of service: Urban Areas


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [18]:
#Type your code here